docs/public/specifications/tests/DNSSEC-TP/dnssec10.md

```diff
@@ -95,183 +107,301 @@ message. The argument names are defined in the [argument list].
+                   2. For each NSEC3 RRSIG do:
+                      1. Verify the RRSIG record by the DNSKEY records.
+                      2. If the Zonemaster installation does not have support for
+                         the DNSKEY algorithm that created the RRSIG, then add
+                         name server IP, DNSKEY algorithm and DNSKEY key tag to
+                         the *Algo Not Supported By ZM* set.
+                      3. Else, if one or more of the following criteria is met,
+                         add the name server IP and the RRSIG key ID to the
+                         *NSEC3 RRSIG Verify Error* set.
+                         1. The RRSIG record has a validity period that starts
+                            after or ends before the time of test execution.
+                         2. There is no DNSKEY that matches RRSIG by key tag.
+                         3. The RRSIG cannot be validated by the DNSKEY record
+                            appointed.
+
+    6. Send *NSEC3PARAM* query to the name server IP and do:
+       1. If at least one of the following criteria is met, then add the name
+          server IP to the *NSEC3PARAM Query Response Error* set:
+          1. There is no DNS response.
+          2. The [RCODE Name] in the response is not "NoError".
+          3. The AA flag is not set in the response.
+       2. Else if the answer section is non-empty, then do:
+          1. Else if the answer section has a NSEC3PARAM RR then add the name
+             server IP to the *NSEC3PARAM In Answer* set.
+          2. Else, then add the name server IP to the
+             *NSEC3PARAM Query Gives Erroneous Answer* set.
+       3. Else if the answer section is empty, then do:
+          1. If the autority section contains no NSEC record then add the name
+             server IP to the *Expected NSEC Missing* set.
+          2. Else do:
+             1. Add the name server IP to the *NSEC3PARAM Query Gives NSEC NODATA* set.
+             2. If the SOA record is missing the authority section then add name
+                server IP to the *NSEC NODATA Missing SOA* set.
+             3. Else if the owner name of SOA record is is not *Child Zone* then
+                add name server IP to the *NSEC NODATA Wrong SOA* set.
+             4. If the authority section contains more than one NSEC record then
+                add name server IP to the *Erroneous Multiple NSEC* set.
+             5. Else do:
+                1. If the owner name of the NSEC record is not *Child Zone* then
+                   add name server IP to the *NSEC Mismatches Apex* set.
+                2. Else if the type list in the NSEC record does not match the
+                   following criteria then add name server IP to the
+                   *NSEC Incorrect Type List* set:
+                   1. Contains SOA, NS, DNSKEY, NSEC and RRSIG.
+                   2. Does not contain NSEC3PARAM or NSEC3.
```

Comment:
Again: the list must contain SOA, NS, DNSKEY, NSEC and RRSIG but neither NSEC3PARAM nor NSEC3?
I’m thinking that it might be clearer to word this paragraph as:

> Else if the type list in the NSEC record matches at least one of the following criteria then add name server IP to the *NSEC Incorrect Type List* set:
> 1. SOA, NS, DNSKEY, NSEC or RRSIG is missing from the list;
> 2. The list contains NSEC3PARAM or NSEC3.